File: themes/localhost/js/server/UserActivityService.js
MetronicApp.factory("UserActivityService", function (JsonServer) {
    return {
        search:function (filter, text, entity, action, importance, options) {
            var _filter = angular.copy(filter, _filter);
            if(_filter){
                if(_filter.fromDate){
                    _filter.fromDate = _filter.fromDate.getTime();
                } else {
                    _filter.fromDate = null;
                }

                if(_filter.toDate){
                    _filter.toDate = _filter.toDate.getTime();
                } else {
                    _filter.toDate = null;
                }
            }

            var result = JsonServer.postByUrl('/csb-modules-service-portlet.useractivity', 'search', {
                text:text, entity:entity, action:action, importance:importance,
                fromDate:_filter.fromDate, toDate:_filter.toDate, start:_filter.first, maxResult:_filter.maxResults,
                serviceContext:{scopeGroupId:scopeGroupId}}, {
                eventName:'UserActivityService.search',
                scope:options ? options.scope : undefined
            });
        }

    }
});

File: market-manager-service/csb-service-api/src/main/java/com/arman/csb/modules/service/UserActivityService.java
package com.arman.csb.modules.service;

import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.exception.SystemException;
import com.liferay.portal.kernel.jsonwebservice.JSONWebService;
import com.liferay.portal.kernel.transaction.Isolation;
import com.liferay.portal.kernel.transaction.Propagation;
import com.liferay.portal.kernel.transaction.Transactional;
import com.liferay.portal.security.ac.AccessControlled;
import com.liferay.portal.service.BaseService;
import com.liferay.portal.service.InvokableService;

/**
 * Provides the remote service interface for UserActivity. Methods of this
 * service are expected to have security checks based on the propagated JAAS
 * credentials because this service can be accessed remotely.
 *
 * @author Brian Wing Shun Chan
 * @see UserActivityServiceUtil
 * @see com.arman.csb.modules.service.base.UserActivityServiceBaseImpl
 * @see com.arman.csb.modules.service.impl.UserActivityServiceImpl
 * @generated
 */
@AccessControlled
@JSONWebService
@Transactional(isolation = Isolation.PORTAL, rollbackFor =  {
    PortalException.class, SystemException.class}
)
public interface UserActivityService extends BaseService, InvokableService {
    /*
     * NOTE FOR DEVELOPERS:
     *
     * Never modify or reference this interface directly. Always use {@link UserActivityServiceUtil} to access the user activity remote service. Add custom service methods to {@link com.arman.csb.modules.service.impl.UserActivityServiceImpl} and rerun ServiceBuilder to automatically copy the method declarations to this interface.
     */

    /**
    * Returns the Spring bean ID for this bean.
    *
    * @return the Spring bean ID for this bean
    */
    public java.lang.String getBeanIdentifier();

    /**
    * Sets the Spring bean ID for this bean.
    *
    * @param beanIdentifier the Spring bean ID for this bean
    */
    public void setBeanIdentifier(java.lang.String beanIdentifier);

    @Override
    public java.lang.Object invokeMethod(java.lang.String name,
        java.lang.String[] parameterTypes, java.lang.Object[] arguments)
        throws java.lang.Throwable;

    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
    public com.liferay.portal.kernel.json.JSONObject search(
        java.lang.String text, java.lang.String entity,
        java.lang.String action, java.lang.String importance,
        java.util.Date fromDate, java.util.Date toDate, int start,
        int maxResult, com.liferay.portal.service.ServiceContext serviceContext)
        throws com.liferay.portal.kernel.json.JSONException;
}
